internal/hcl/modules/loader.go

```diff
@@ -452,6 +463,22 @@ func findGitRepoRoot(startPath string) (string, error) {
 	return strings.TrimSpace(string(output)), nil
 }
 
+// isSparseCheckoutEnabled checks if sparse-checkout is enabled in the repository
+func isSparseCheckoutEnabled(repoRoot string) (bool, error) {
+	cmd := exec.Command("git", "config", "--get", "core.sparseCheckout")
+	cmd.Dir = repoRoot
+	output, err := cmd.Output()
+	if err != nil {
+		// if exit status is 1, then the config is not set
+		if exitErr, ok := err.(*exec.ExitError); ok && exitErr.ExitCode() == 1 {
+			return false, nil
+		}
```

Comment:
Nitpick, feel free to ignore :D
```suggestion
        var exitErr *exec.ExitError
        if errors.As(err, &exitErr) && exitErr.ExitCode() == 1 {
            return false, nil
        }
```